frontend/src/components/pages/Settings.tsx

```diff
@@ -38,26 +46,97 @@ const PLACEHOLDER_MOBILE_EXAMPLE_ADDRESS = "Address of organization";
 const PLACEHOLDER_MOBILE_EXAMPLE_ORG_DESCRIPTION =
   "Description of organization";
 
+const UPDATEUSERBYID = gql`
+  mutation UpdateUserById(
+    $requestorId: String!
+    $id: String!
+    $userInfo: UserInfoInput!
+  ) {
+    updateUserByID(requestorId: $requestorId, id: $id, userInfo: $userInfo) {
+      user {
+        id
+        info {
+          email
+          organizationAddress
+          organizationName
+          organizationDesc
+          role
+          roleInfo {
+            aspInfo {
+              numKids
+            }
+            donorInfo {
+              type
+              tags
+            }
+          }
+          primaryContact {
+            name
+            phone
+            email
+          }
+          onsiteContacts {
+            name
+            phone
+            email
+          }
+        }
+      }
+    }
+  }
+`;
+
 const Settings = (): React.ReactElement => {
-  const [primaryContact, setPrimaryContact] = useState<Contact>({
-    name: "",
-    phone: "",
-    email: "",
-  });
-  const [organizationName, setOrganizationName] = useState("");
-  const [numberOfKids, setNumberOfKids] = useState("");
-  const [organizationAddress, setOrganizationAddress] = useState("");
-  const [organizationDescription, setOrganizationDescription] = useState("");
-  const [onsiteInfo, setOnsiteInfo] = useState<Array<Contact>>([
-    {
+  // Assumption: user has the roleInfo: ASPInfo
+  const { authenticatedUser, setAuthenticatedUser } = useContext(AuthContext);
+  const userInfo: UserInfo = authenticatedUser?.info || null;
+
+  const [primaryContact, setPrimaryContact] = useState<Contact>(
+    userInfo?.primaryContact || {
       name: "",
       phone: "",
       email: "",
     },
-  ]);
+  );
+  const [organizationName, setOrganizationName] = useState(
+    userInfo?.organizationName || "",
+  );
+  const [numKids, setNumKids] = useState(
+    userInfo?.roleInfo?.aspInfo?.numKids?.toString() || "",
+  );
+  const [organizationAddress, setOrganizationAddress] = useState(
+    userInfo?.organizationAddress || "",
+  );
+  const [organizationDesc, setOrganizationDesc] = useState(
+    userInfo?.organizationDesc || "",
+  );
+  // json parse/stringify creates a deep copy of the array of contacts
+  const [onsiteInfo, setOnsiteInfo] = useState<Array<Contact>>(
+    userInfo
+      ? JSON.parse(JSON.stringify(userInfo.onsiteContacts))
```

Comment:
need to deep clone `onsiteContacts` to prevent `setOnsiteInfo` from mutating the default values of `userInfo.onsiteContacts`